core/tests/protractor/learnerDashboard.js

```diff
@@ -129,17 +133,15 @@ describe('Learner dashboard functionality', function() {
     // 'In Progress' section.
     player.submitAnswer('Continue', null);
     browser.ignoreSynchronization = true;
-    browser.get(general.LEARNER_DASHBOARD_URL);
+    learnerDashboardPage.get();
     general.acceptAlert();
     browser.waitForAngular();
     general.waitForSystem();
     browser.ignoreSynchronization = false;
-    element(by.css('.protractor-test-incomplete-collection-section')).click();
+    learnerDashboardPage.navigateToIncompleteCollectionsSection();
     browser.waitForAngular();
```

Comment:
It might be better to put these browser.waitForAngular() calls at the end of their respective functions in the page object, if  they're always going to be called. This will also help simplify the main protractor test.

Ditto for general.waitForSystem(), where relevant.

Reply:
If there are are other calls to learnerDashboardPage that are repeating for all such instances, shall I put it inside page object also?

Reply:
Sorry, I don't understand. Could you elaborate / give an example?

Reply:
after both instances of navigateToIncompleteCollectionsSection(), expectTitleOfSummaryTileToMatch() is called, so can I put this call as this.expectTitleOfSummaryTileToMatch() inside the navigateToIncompleteCollectionsSection() function?

Reply:
Ah, I see. No, don't do that.

The way to think about this is that these are the actions a user would do. A user wouldn't really be doing stuff at the level of "wait for angular". But they may take the actions "navigate" and "expect title to match" independently, so these should be independent actions in the page object.

Reply:
Ok, thanks.

Reply:
Done